middleman-core/lib/middleman-core/preview_server.rb

```diff
@@ -19,13 +21,27 @@ def https?
       # Start an instance of Middleman::Application
       # @return [void]
       def start(opts={})
+        # Otherwise testing becomes hard
+        $stdout.sync = true
+        $stderr.sync = true
+
         @options = opts
+        @server_information = ServerInformation.new
+
+        the_app = new_app
```

Comment:
This evaluates the middleman configuration so I needed to move it to a variable and place it before the warning.